src/modules/75-cf/pages/environment-details/EnvironmentDetails.tsx

```diff
@@ -5,54 +5,124 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
-import React from 'react'
+import React, { useState, useCallback } from 'react'
 import { useParams } from 'react-router-dom'
-import { Layout, Container, PageError } from '@wings-software/uicore'
+import { PageError } from '@harness/uicore'
+import routes from '@common/RouteDefinitions'
 import { ContainerSpinner } from '@common/components/ContainerSpinner/ContainerSpinner'
 import { useDocumentTitle } from '@common/hooks/useDocumentTitle'
 import type { EnvironmentResponseDTO } from 'services/cd-ng'
+import { ApiKey, useGetAllAPIKeys } from 'services/cf'
 import { useSyncedEnvironment } from '@cf/hooks/useSyncedEnvironment'
 import { useStrings } from 'framework/strings'
-import { getErrorMessage } from '@cf/utils/CFUtils'
-import CFEnvironmentDetailsHeader from './EnvironmentDetailsHeader'
+import { CF_DEFAULT_PAGE_SIZE, getErrorMessage } from '@cf/utils/CFUtils'
+import AddKeyDialog from '@cf/components/AddKeyDialog/AddKeyDialog'
+import { EnvironmentType } from '@common/constants/EnvironmentType'
+import useActiveEnvironment from '@cf/hooks/useActiveEnvironment'
+import { DetailPageTemplate } from '@cf/components/DetailPageTemplate/DetailPageTemplate'
 import CFEnvironmentDetailsBody from './EnvironmentDetailsBody'
 
 const EnvironmentDetails: React.FC = () => {
   const { getString } = useStrings()
   const { projectIdentifier, environmentIdentifier, orgIdentifier, accountId } = useParams<Record<string, string>>()
+  const { withActiveEnvironment } = useActiveEnvironment()
+  const [page, setPage] = useState<number>(0)
+  const [recents, setRecents] = useState<ApiKey[]>([])
+
+  const updateApiKeyList = (key: ApiKey): void => {
+    setRecents([...recents, key])
+  }
+
+  const queryParams = {
+    projectIdentifier: projectIdentifier,
+    environmentIdentifier: environmentIdentifier,
+    accountIdentifier: accountId,
+    orgIdentifier: orgIdentifier
+  }
```

Comment:
You can get some auto-complete going by typing this
```suggestion
  const queryParams: GetAllAPIKeysQueryParams = {
    projectIdentifier,
    environmentIdentifier,
    accountIdentifier: accountId,
    orgIdentifier
  }
```

Totally optional